File: src/main/java/com/langleon/dsobuildsim/jewels/dto/JewelDefinitionDTO.java
package com.langleon.dsobuildsim.jewels.dto;

import com.langleon.dsobuildsim.common.StatType;
import com.langleon.dsobuildsim.jewels.JewelType;

import java.util.Map;

public record JewelDefinitionDTO(JewelType jewelType, String name, int tier, Map<Integer, Map<StatType, Double>> statsPerTier, Map<Integer, String> descriptionPerTier) {
}

File: frontend/src/app/models/gamedataDTOs/JewelDefinitionDTO.ts
import { StatType } from "../../enums/StatType";

export interface JewelDefinitionDTO {
    jewelType: string;
    jewelLimitGroup: string;
    name: string;
    tier: number;
    statsPerTier: Record<string, Record<StatType, number>>;
    descriptionPerTier: Record<string, string>;
}